odoo/models.py

```diff
@@ -4366,9 +4379,12 @@ def ensure_one(self):
         """ Verifies that the current recorset holds a single record. Raises
         an exception otherwise.
         """
-        if len(self) == 1:
+        try:
+            # unpack to ensure there is only one value is faster than len for big lists
```

Comment:
I find this comment misleading.

Seems to me the new solution is indeed slighly faster when comparing a single record than on failing.
But then, I found the comment misleading. What is faster, is calling this method with valid len of 1, not with a big list of ids.
Otherwise, it seems it make it slower when it fails. Which is fine, as we want to be fast when the len is correct and which is the main use case.

```suggestion
            # unpack to ensure there is only one value is faster than len when true and
            # has a significant impact as this check is largely called
```

``` python
>>> s1 = 'def s1(l):\n    if len(l) == 1:\n        return True\n    return False\ns1(a)\n'
>>> s2 = 'def s2(l):\n    try:\n        x, = l\n        return True\n    except ValueError:\n        return False\ns2(a)\n'
>>> a = [1]
>>> timeit.Timer(s1, 'from __main__ import a').timeit()
0.1974480152130127
>>> timeit.Timer(s2, 'from __main__ import a').timeit()
0.17123007774353027
>>>
>>> a = range(10**6)
>>> timeit.Timer(s1, 'from __main__ import a').timeit()
0.21253204345703125
>>> timeit.Timer(s2, 'from __main__ import a').timeit()
1.283552885055542
>>> 
>>> a = []
>>> timeit.Timer(s1, 'from __main__ import a').timeit()
0.20595693588256836
>>> timeit.Timer(s2, 'from __main__ import a').timeit()
1.6445159912109375
```

Reply:
I agree the comment is misleading. It doesn't matter if the list is big or not
```Python
%timeit id, = [1]
```
~9.0 ns
```Python
%timeit len([1]) == 1
```
~60 ns

It's just that a the time I wrote the patch, I just noticed this was faster and I thought this was because of the list had a bunch of items. After a deeper investigation, I found out it was just the inference of calling a method (whatever if was `len` or anything else), as the unpack was directly converted to opcode, while the `len`wasn't.

I will change the comment.